artist/src/main/kotlin/com/uber/artist/ArtistPlugin.kt

```diff
@@ -73,6 +74,13 @@ class ArtistPlugin : Plugin<Project> {
       artistTask.outputs.dir(outputDir)
       generateViews.dependsOn(artistTask)
 
+      if (artistExtension.generateKotlin) {
+        val kotlinCompileTask = project.tasks.findByName("compile${variant.name.capitalize()}Kotlin")
+        if (kotlinCompileTask != null) {
+          generateViews.dependsOn(kotlinCompileTask)
+        }
+      }
```

Comment:
I believe this is backwards. You need the KotlinCompile task to depend on its outputs. That said, the below line (`registerJavaGeneratingTask(...)`) should be handling this for you. If it's not working, then I think there's something else going wrong or you've possibly hit a bug.

Reply:
Yes, this is backward! And something fishy is happening anyway : the output folder is not being added to the sourceset (even though it's registered). I assume it's working for the JAVA case, and may be only an issue with KT? The current workaround is to add the (hardcoded) generated folder path from build.gradle, but should be fixed.

Reply:
Possibly. The project is on a pretty old version of AGP though, maybe try updating to the latest stable (4.1.2) and see if it's resolved there. Definitely a lot of kotlin improvements have come over the past couple years

Reply:
I am on 4.1.20 already :)

Reply:
Cleaned this up in #39 so that it is less confusing going forward